modules/loader/source/relocations.cpp

```diff
@@ -832,6 +841,49 @@ bool resolveRISCV(const loader::Relocation &r, loader::ElfMap &map,
       writeITypeImm(getLo12(relative_value), relocation_address + 4);
       break;
     }
+    case R_RISCV_BRANCH: {
+      // R_RISCV_BRANCH immediate is for B-Type instructions and is split
+      // across 12 bits spread across the instruction and skips the bottom bit
+      // see conditional branches in the isa spec
+
+      if (signExtendN(relative_value, 13) !=
+          static_cast<int64_t>(relative_value & (~1))) {
+        CARGO_ASSERT(0, "R_RISCV_BRANCH branch > 13 bits or not even.");
+        return false;
+      }
+      uint32_t value;
+      const uint32_t trunc_value = static_cast<uint32_t>(relative_value);
+      cargo::read_little_endian(&value, relocation_address);
+      // imm bits 1-4 at bit 8
+      value = setBitRange(value, trunc_value >> 1, 8, 4);
+      // imm bits 5-10 at bit 25
+      value = setBitRange(value, trunc_value >> 5, 25, 6);
+      // imm bit 11 at bit 7
+      value = setBitRange(value, trunc_value >> 11, 7, 1);
+      // imm bit 12 at bit 31
+      value = setBitRange(value, trunc_value >> 12, 31, 1);
+      cargo::write_little_endian(value, relocation_address);
+      break;
+    }
+
+    case R_RISCV_RVC_JUMP: {
+      // R_RISCV_RVC_JUMP is for compressed instructions and fits in bits 2-12
+      // of a 16 bit value and skips the bottom bit. See CJ format in the isa
+      // spec.
+      if (signExtendN(relative_value, 13) !=
+          static_cast<int64_t>(relative_value & (~1))) {
```

Comment:
This check does not match what it says it checks. This checks that the vaue fits in 13 bits (and that the bottom bit is zero) but the below assert says it needs to fit in 12 bits. I *think* the text is correct and this should be
```suggestion
      if (signExtendN(relative_value, 12) !=
          static_cast<int64_t>(relative_value & (~1))) {
```
but please double-check, I could be wrong.

Reply:
It's 13 bits because the bottom bit gets dropped. I could also have shifted it by one  but then couldn't sneak in the ~1 bit.

Reply:
I'll add some comments to make this clear

Reply:
yes you are right, this one is different.